site/documentation/content/admin-guide/mongodb-device-registry-config.md

```diff
@@ -0,0 +1,90 @@
+| `HONO_CREDENTIALS_SVC_COLLECTION_NAME`<br>`--hono.credentials.svc.collectionName` | no | `credentials` | The name of the MongoDB collection where the server stores credentials of devices.|
+| `HONO_CREDENTIALS_SVC_HASH_ALGORITHMS_WHITELIST`<br>`--hono.credentials.svc.hashAlgorithmsWhitelist` | no | `empty` | An array of supported hashing algorithms to be used with the `hashed-password` type of credentials. When not set, all values will be accepted. |
+| `HONO_CREDENTIALS_SVC_MAX_BCRYPT_ITERATIONS`<br>`--hono.credentials.svc.maxBcryptIterations` | no | `10` | The maximum number of iterations that are supported in password hashes using the BCrypt hash function. This limit is enforced by the device registry when adding or updating corresponding credentials. Increasing this number allows for potentially more secure password hashes to be used. However, the time required to compute the hash increases exponentially with the number of iterations. |
+| `HONO_CREDENTIALS_SVC_MODIFICATION_ENABLED`<br>`--hono.credentials.svc.modificationEnabled` | no | `true` | When set to `false` the credentials contained in the registry cannot be updated nor removed. |
+| `HONO_CREDENTIALS_SVC_RECEIVER_LINK_CREDIT`<br>`--hono.credentials.svc.receiverLinkCredit` | no | `100` | The number of credits to flow to a client connecting to the Credentials endpoint. |
+| `HONO_DEVICE_CONNECTION_SVC_MAX_DEVICES_PER_TENANT`<br>`--hono.deviceConnection.svc.maxDevicesPerTenant` | no | `100` | The number of devices per tenant for which connection related data is stored. It is an error to set this property to a value <= 0. |
+| `HONO_MONGODB_CONNECTION_STRING`<br>`--hono.mongodb.connectionString` | no | - | The connection string used by the Device Registry application to connect to the MongoDB database. If `HONO_MONGODB_CONNECTION_STRING` is set, it overrides the other MongoDB connection settings.<br> See [Connection String URI Format](https://docs.mongodb.com/manual/reference/connection-string/) for more information.|
+| `HONO_MONGODB_CONNECTION_TIMEOUT_IN_MS`<br>`--hono.mongodb.connectionTimeoutInMs` | no | `10000` | The time in milliseconds to attempt a connection before timing out.|
+| `HONO_MONGODB_DB_NAME`<br>`--hono.mongodb.dbName` | no | - | The name of the MongoDB database that should be used by the Device Registry application. |
+| `HONO_MONGODB_HOST`<br>`--hono.mongodb.host` | no | `localhost` | The host name or IP address of the MongoDB instance.|
+| `HONO_MONGODB_PORT`<br>`--hono.mongodb.port` | no | `27017` | The port that the MongoDB instance is listening on.|
+| `HONO_MONGODB_PASSWORD`<br>`--hono.mongodb.password` | no | - | The password to use for authenticating to the MongoDB instance.|
+| `HONO_MONGODB_SERVER_SELECTION_TIMEOUT_IN_MS`<br>`--hono.mongodb.serverSelectionTimeoutInMs` | no | `300000` | The time in milliseconds that the mongo driver will wait to select a server for an operation before raising an error.|
```

Comment:
What implication does the default value have? Does this mean that an operation will not fail before 300s have passed? If so, I do not think that this is a reasonable default value.

Reply:
If it is a stand-alone server, there is normally only one server available and it gets selected. In a shared clusters or replica set,  MongoDB drivers use a server selection algorithm to choose which one to connect to and for that the server selection time out is being used.

My bad, it is a typo in the JavaDoc and for the documentation I referred the JavaDoc. The default value is 30 _seconds_. I am updating it.